sunpy/coordinates/tests/test_transformations.py

```diff
@@ -116,3 +116,16 @@ def test_hgs_hgc_roundtrip():
 
     assert_quantity_allclose(hgsout.lat, hgsin.lat)
     assert_quantity_allclose(hgsout.lon, hgsin.lon)
+
+
+def test_dateobs_hgs_hgc_roundtrip():
+    obstime = "2011-01-01"
+
+    hgsin = HeliographicStonyhurst(lat=0*u.deg, lon=0*u.deg, obstime=obstime)
+    hgcout = hgsin.transform_to(HeliographicCarrington)
+
+    assert hgsin.obstime == hgcout.obstime
+
+    hgsout = hgcout.transform_to(HeliographicStonyhurst)
+
+    assert hgsout.obstime == hgsin.obstime
```

Comment:
So we might have a bug here (not in your code, but in SunPys).

```hgsout.obstime``` seems to be missing, which is why the test is failing. I will speak to some people about how we want to handle this.